src/trackPlayer.ts

```diff
@@ -62,8 +63,8 @@ function registerPlaybackService(factory: () => ServiceHandler) {
   }
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function addEventListener(event: Event, listener: (data: any) => void) {
+type EventListener<T extends Event> = (data: EventListenerProp[T]) => void
```

Comment:
We need to ensure that this is a backward-compatible change. Meaning, we shouldn't require an event type to be passed (needs a default value). People should be able to opt-into this if they want the type safety of it but not be required to update their code when they upgrade to the newest version.

Reply:
Ah yeah, this one might be a bit tricky.. the change is nice! but it will start throwing typescript errors for those that update.. any ideas on things we can do @Spice-Z ?